docs/concepts/model_endpoints.md

```diff
@@ -0,0 +1,106 @@
+# Model Endpoints
+
+Model Endpoints are deployments of models that can receive requests and return
+predictions containing the results of the model's inference. Each model endpoint
+is associated with a model bundle, which contains the model's code. An endpoint
+specifies deployment parameters, such as the minimum and maximum number of
+workers, as well as the requested resources for each worker, such as the number
+of CPUs, amount of memory, GPU count, and type of GPU.
+
+Endpoints can be asynchronous or synchronous. Asynchronous endpoints return
+a future immediately after receiving a request, and the future can be used to
+retrieve the prediction once it is ready. Synchronous endpoints return the
+prediction directly after receiving a request.
+
+## Creating Async Model Endpoints
+
+Async model endpoints are the most cost-efficient way to perform inference on
```

Comment:
While we're here, might be worth doing a small content refresh. My thinking is that we can lift this paragraph and the corresponding one under `Creating Sync Endpoints` and centralize them into the top-level `Model Endpoints` section, or have a dedicated sub-section explaining the tradeoffs.

Would also suggest adding a note about the tradeoffs w.r.t responsibilities of queue management to supplement the existing highlighted tradeoff between latency and cost.

Reply:
<img width="837" alt="image" src="https://user-images.githubusercontent.com/92065453/222317276-ee3f73e3-a5cf-4365-9c81-db37ee1c9c6c.png">